contracts/contracts/GrantRegistry.sol

```diff
@@ -1,135 +1,124 @@
 pragma solidity ^0.8.5;
 
 contract GrantRegistry {
-  // Grant ID - The hash of the grants meetadata
-  bytes32 public grantRegistryId;
-  // Metadata Pointer
-  string public metaPtr;
-
-  // Owner
-  address owner;
-
-  modifier onlyOwner() {
-    require(msg.sender == owner);
-    _;
-  }
-
-  // TODO: Only allow the owner (which is the curator) address to configure the contract
-  // while its in config mode, then switch to active. Once active, the owner can only
-  // close the registry (have a conversation around this approach)
-  constructor(
-    address _owner,
-    bytes32 _grantRegistryId,
-    string memory _metaPtr
-  ) {
-    owner = _owner;
-    //state = State.CONFIG;
-    grantRegistryId = _grantRegistryId;
-    metaPtr = _metaPtr;
-  }
-
-  // State
-  enum State {
-    CONFIG,
-    ACTIVE,
-    CLOSED
-  }
-  //State public state;
-
-  // Grant Pointer Object
-  // Mapping from GrantID to set of Owner Address, Payout Address, Metadata Pointer, Replaces Grant (optional)
-  // TODO: Implement safemath for avoiding overflows with ints
-  uint256 public grantCount = 0;
-  mapping(bytes32 => Grant) public grants;
-  bytes32[] public grantIds;
+  // --- Data ---
+  /// @notice Number of grants stored in this registry
+  uint96 public grantCount;
 
+  /// @notice Grant object
   struct Grant {
-    bytes32 id;
-    address owner;
-    address payout;
-    string metaPtr;
-    State state;
-    bytes32 replaces;
+    uint96 id; // grant ID, as uint96 to pack into same slot as owner (this implies a max of 2^96-1 = 7.9e28 grants)
+    address owner; // grant owner (has permissions to modify grant information)
+    address payee; // address that receives funds donated to this grant
+    string metaPtr; // URL pointing to grant metadata (for off-chain use)
   }
 
-  function activate(bytes32 _id) external onlyOwner {
-    grants[_id].state = State.ACTIVE;
-  }
+  /// @notice Mapping from Grant ID to grant data
+  mapping(uint96 => Grant) public grants; // TODO use an array instead with ID to index it? Which is better? Will array copy full array to memory in `getAllGrants`?
 
-  function closed(bytes32 _id) public onlyOwner {
-    grants[_id].state = State.CLOSED;
-  }
+  // --- Events ---
+  /// @notice Emitted when a new grant is created
+  event GrantCreated(uint96 indexed id, address indexed owner, address indexed payee, string metaPtr);
+
+  /// @notice Emitted when a grant's owner is changed
+  event GrantUpdated(uint96 indexed id, address indexed owner, address indexed payee, string metaPtr);
```

Comment:
Wondering if we should include the old value in the update event?

Reply:
I was thinking about that, but decided against it because of gas savings, and if you need the old values they are still available through other logs. So I couldn't think of a strong reason for including 3 additional parameters

Reply:
Sounds good. Let's leave it out for now, but as we're working on the frontend, we can think about if we hit any places where we'd wish we had the old values in the event.